drivers/hud_spm_report/app/models/hud_spm_report/fy2023/spm_enrollment.rb

```diff
@@ -256,25 +261,42 @@ def self.detail_headers
       if enrollment.exit.present? && enrollment.exit.exit_date <= end_date
         enrollment.income_benefits_at_exit
       else
-        enrollment.
-          income_benefits_annual_update.
-          where(information_date: ..end_date).
-          where(annual_update_window_sql(enrollment)).
-          order(information_date: :desc).
-          first
+        # enrollment.
+        #   income_benefits_annual_update.
+        #   where(information_date: ..end_date).
+        #   where(annual_update_window_sql(enrollment)).
+        #   order(information_date: :desc).
+        #   first
+        enrollment.income_benefits.select do |ib|
+          ib.data_collection_stage == 5 && ## Annual update
+            ib.information_date <= end_date &&
+            date_in_annual_update_window(ib.information_date, enrollment)
+        end.sort_by(&:information_date).reverse.first
       end
     end
 
     private_class_method def self.previous_income_benefits(enrollment, annual_date, end_date)
       return enrollment.income_benefits_at_entry if enrollment.exit.present? && enrollment.exit.exit_date <= end_date
 
       # Most recent annual update on or before the renewal date, or the entry assessment
-      enrollment.
-        income_benefits_annual_update.
-        where(information_date: ...annual_date).
-        where(annual_update_window_sql(enrollment)).
-        order(information_date: :desc).
-        first || enrollment.income_benefits_at_entry # Default to entry assessment if less than 2 years
+      # enrollment.
+      #   income_benefits_annual_update.
+      #   where(information_date: ...annual_date).
+      #   where(annual_update_window_sql(enrollment)).
+      #   order(information_date: :desc).
+      enrollment.income_benefits.select do |ib|
+        ib.data_collection_stage == 5 && ## Annual update
+          ib.information_date < annual_date &&
+          date_in_annual_update_window(ib.information_date, enrollment)
+      end.sort_by(&:information_date).reverse.first || enrollment.income_benefits_at_entry # Default to entry assessment if less than 2 years
+    end
+
+    private_class_method def self.date_in_annual_update_window(date, enrollment)
+      entry_date = enrollment.entry_date
+      interval = 30.days
+      window_date = Date.new(date.year, entry_date.month, entry_date.day)
+
+      window_date - interval .. window_date + interval
```

Comment:
This appears to be returning the range rather than the boolean.  Seems like it should be returning a true or false "in the window".  Also, maybe we should add a `?` to the method name to indicate that?